sandbox/libs/authn/src/main/java/org/opensearch/authn/User.java

```diff
@@ -10,17 +10,20 @@
 import java.util.Collections;
 import java.util.Map;
 
-public class User {
+public class User implements Subject {
 
     @JsonProperty(value = "primary_principal")
     private StringPrincipal primaryPrincipal;
 
     @JsonProperty(value = "hash")
     private String bcryptHash;
+
+    @JsonProperty(value = "attributes")
     private Map<String, String> attributes = Collections.emptyMap();
 
+    @Override
     @JsonProperty(value = "primary_principal")
-    public StringPrincipal getPrimaryPrincipal() {
+    public StringPrincipal getPrincipal() {
```

Comment:
The previous version implied that a subject could have more than one principal (which is true), this one does not imply this. Is this intentional or was this change made since it should only ever be interacting with a specific principal at a time?